util/canonical.go

```diff
@@ -101,6 +104,14 @@ func ConnectionFromCanonical(s string) *Connection {
 	}
 }
 
+func truncateIPv6Bracket(ip string) string {
```

Comment:
I am not sure but maybe the `net.SplitHostPort` can help and also `ConnectionFromCanonical` is redundant. Could you check?